src/tools/clippy/clippy_lints/src/redundant_pub_crate.rs

```diff
@@ -52,7 +53,7 @@ impl<'tcx> LateLintPass<'tcx> for RedundantPubCrate {
             && is_not_macro_export(item)
             && !item.span.in_external_macro(cx.sess().source_map())
         {
-            let span = item.span.with_hi(item.ident.span.hi());
+            let span = item.span.with_hi(item.kind.ident().unwrap_or(Ident::empty()).span.hi());
```

Comment:
Maybe change it to the following temporarily to make it even clearer that this is butchered (apart from the FIXME comment I suggested):

```suggestion
            let span = item.span.with_hi(item.kind.ident().map(|ident| ident.span.hi()).unwrap_or(DUMMY_SP.hi()));
```

Since this is line is about "span computation" not anything Ident related, using `Ident::empty` or `Ident::dummy` is but a detour I'd say.